src/bankid.ts

```diff
@@ -33,9 +37,22 @@ export interface AuthResponse {
 interface AuthOptionalRequirements {
   cardReader?: "class1" | "class2";
   certificatePolicies?: string[];
-  issuerCn?: string[];
-  autoStartTokenRequired?: boolean;
-  allowFingerprint?: boolean;
+  mrtd?: boolean;
+  personalNumber?: string;
+  pinCode?: boolean;
+}
+
+interface AppData {
+  appIdentifier: string;
+  deviceOS: string;
+  deviceModelName: string;
+  deviceIdentifier: string;
+}
+
+interface WebData {
+  deviceIdentifier: string;
+  referringDomain: string;
+  userAgent: string;
 }
```

Comment:
Could we omit the `Data` from the type and map it 1-1 with bankid?

In the [payment spec](https://developers.bankid.com/api-references/auth--sign/payment#request-body) they mention that:

  > You can send the parameter web or app, not both. If providing this parameter, at least one of its members must be specified.

  Can we create a union to not allow both being specified at the same time?

  To achieve "at least one of its member must be specified" we could do something like this:

```suggestion
type AtLeastOne<T> = {
  [K in keyof T]: Required<Pick<T, K>> & Partial<Omit<T, K>>;
}[keyof T];

type App = AtLeastOne<{
  appIdentifier: string;
  deviceOS: string;
  deviceModelName: string;
  deviceIdentifier: string;
}>;

type Web = AtLeastOne<{
  deviceIdentifier: string;
  referringDomain: string;
  userAgent: string;
}>;
```